src/Lecture1.hs

```diff
@@ -98,7 +104,11 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int -> Int -> String -> String
+subString start end
+  | end < 0 = const []
+  | start < 0 = subString 0 end
```

Comment:
You can use `take` here directly to avoid an extra recursive call and an extra call to `drop`
```suggestion
  | start < 0 = take (end + 1)
```